inference-engine/src/inference_engine/compilation_context.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "compilation_context.hpp"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+#include <xml_parse_utils.h>
+
+#include "ie_itt.hpp"
+#include "cpp_interfaces/exception2status.hpp"
+#include "transformations/serialize.hpp"
+#include "cnn_network_ngraph_impl.hpp"
+#include "cpp/ie_cnn_network.h"
+#include "details/ie_exception.hpp"
+
+#include "ngraph/pass/pass.hpp"
+#include "ngraph/variant.hpp"
+#include "ngraph/function.hpp"
+#include "ngraph/opsets/opset6.hpp"
+
+#ifdef WIN32
+#define stat _stat
+#endif
+
+namespace InferenceEngine {
+
+template <typename T>
+static std::size_t hash_combine(std::size_t seed, const T& a) {
+    std::size_t val = std::hash<T>()(a);
+
+    // Hash combine formula from boost
+    return seed ^ (val + 0x9e3779b9 + (seed << 6) + (seed >> 2));
+}
+
+template <typename T>
+static int32_t as_int32_t(T v) {
+    return static_cast<int32_t>(v);
+}
+
+class OstreamHashWrapper final: public std::streambuf {
+    std::size_t    m_res = {};
+    std::streampos m_pos = {};
+
+public:
+    std::size_t getResult() const { return m_res; }
+    std::streamsize xsputn(const char* s, std::streamsize n) override {
+        std::string str(s, n);
```

Comment:
here we create tmp variable from ngraph constants, if constant is big enough, we still increase memory consumption.